tform/aws-resources/main.tf

```diff
@@ -0,0 +1,70 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.27"
+    }
+    github = {
+      source  = "integrations/github"
+      version = "~> 4.0"
+    }
+  }
+
+  required_version = ">= 0.14.9"
+}
+
+locals {
+  owner = "osbuild"
+  repo  = "automotive-ci"
+
+  tags = {
+    owner = "terraform"
+    team  = "a-team"
+    env   = "prod"
+  }
+}
+
+
+provider "aws" {
+  profile = "default"
+  region  = "eu-west-1"
+}
+
+
+provider "github" {
+  owner = local.owner
+
+}
+
+
+resource "aws_s3_bucket" "auto-ci-pipeline-repos" {
+  bucket = "auto-ci-pipeline-repos"
+  acl    = "private"
+
+  tags = local.tags
+
+}
+
+
+resource "aws_s3_bucket" "auto-ci-pipeline-images" {
+  bucket = "auto-ci-pipeline-images"
```

Comment:
The repo name should be consumed directly from Actions secret. Avoid hard coding it.

Reply:
I think that the GitHub resources should be created first and then consumed by the AWS resources.

Reply:
In this case, it's not possible, we need to create the bucket first before we can populate the secret with the bucket arn

Reply:
Or populate the secret with the hard coded value and then consume the value when creating the bucket.

Reply:
So just to clarify, do we want to consume the bucket name from the github secret, create it and repopulate another secret with the arn?

Reply:
Where is the arn used? I am not following

Reply:
Isn't the bucket arn and the name have the same value?

Reply:
Sorry, my bad, it is the bucket name that's in the secret, I must have confused myself (and as a result you as well) with what's in the secret. You're right it is the bucket name

Reply:
In which case we don't need to repopulate the secret with the bucket name at the end of the plan